lib/travis/services/find_admin.rb

```diff
@@ -29,25 +29,33 @@ def repository
       private
 
         def candidates
-          User.with_github_token.with_permissions(:repository_id => repository.id, :admin => true)
+          User.with_github_token.with_permissions(:repository_id => repository.id, :admin => true).first(NUM_CANDIDATES)
+        end
+
+        def admin
+          admin = candidates.detect do |candidate|
+            is_valid? candidate
+          end
+          admin || raise_admin_missing
         end
 
-        def validate(user)
+        def is_valid?(admin)
           Timeout.timeout(2) do
-            data = Github.authenticated(user) { repository_data }
+            data = Github.authenticated(admin) { repository_data }
             if data['permissions'] && data['permissions']['admin']
-              user
+              true
             else
-              info "[github-admin] #{user.login} no longer has admin access to #{repository.slug}"
-              update(user, data['permissions'])
+              info "[github-admin] #{admin.login} no longer has admin access to #{repository.slug}"
+
+              admin.update_attributes!(:permissions => { :admin => false })
```

Comment:
True, `update_attributes!` alone wouldn't update the `permission` record.

Fixed and also fixed tests.